libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRoutePreviewScreen.kt

```diff
@@ -34,37 +34,38 @@ import com.mapbox.maps.MapboxExperimental
 import com.mapbox.maps.extension.androidauto.MapboxCarMapObserver
 import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
 import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
 
 /**
  * After a destination has been selected. This view previews the route and lets
  * you select alternatives. From here, you can start turn-by-turn navigation.
  */
 @MapboxExperimental
+@ExperimentalPreviewMapboxNavigationAPI
 class CarRoutePreviewScreen(
     private val routePreviewCarContext: RoutePreviewCarContext,
     private val placeRecord: PlaceRecord,
     private val navigationRoutes: List<NavigationRoute>,
     private val placesLayerUtil: PlacesListOnMapLayerUtil = PlacesListOnMapLayerUtil(),
 ) : Screen(routePreviewCarContext.carContext) {
 
-    private val routesProvider = PreviewRoutesProvider(navigationRoutes)
```

Comment:
Why did you remove this? We don't want to set routes during route preview (too many issues with this approach), thus we need to store the routes locally.

Reply:
The `RouteLineComponent` will only use routes set by `MapboxNavigation`. There is also an issue with not setting the route during route preview, route progress does not work. So if you try navigating during route preview it will go stale.

But yeah I'll look to update this. RouteLineComponent will need to be able to accept the route outside of MapboxNavigation

Reply:
I'm making this change blocked by https://github.com/mapbox/mapbox-navigation-android/issues/5990

Reply:
> too many issues with this approach

Is there a list of issues somewhere? I don't know how you could make route preview a state of active navigation otherwise

Reply:
> I don't know how you could make route preview a state of active navigation otherwise

Why do we want to do this? Is there an actual customer request to do so?

Reply:
I would not worry about route progress not working in route preview state because that is something that should not happen. Route preview allows me to decide the route I want to choose before actually starting the navigation. In that case, why would I want `routeProgress` in Route Preview mode.

Reply:
What's the reason for implementing >I'm making this change blocked by https://github.com/mapbox/mapbox-navigation-android/issues/5990

in this PR? Can we keep the implementation separate so that we don't make the PR complicated and keep the solution to what the original ticket is asking for?

Reply:
Android Auto cannot use RouteLineComponent for the same reason that 1tap cannot use RouteLineComponent, as Zayankovsky is warning by the first comment in the thread.

RoutePreview cannot set the MapboxNavigation route. But the RouteLineComponent is essentially an `ActiveGuiidanveRouteLineComponent`